controller/scheduler/fleet.py

```diff
@@ -271,14 +271,14 @@ def run(self, name, image, entrypoint, command):  # noqa
             tran = ssh.get_transport()
 
             def _do_ssh(cmd):
-                chan = tran.open_session()
-                # get a pty so stdout/stderr look right
-                chan.get_pty()
-                out = chan.makefile()
-                chan.exec_command(cmd)
-                output = out.read()
-                rc = chan.recv_exit_status()
-                return rc, output
+                with tran.open_session() as chan:
+                    chan.exec_command(cmd)
+                    while not chan.exit_status_ready():
+                        time.sleep(1)
```

Comment:
Instead of infinitely sleeping here in case of a timeout, should we implement [this](https://github.com/paramiko/paramiko/blob/54415207d186155afab6bd3a8571a70587a85fab/tests/test_transport.py#L349-L353) here so we at least perform a timeout after 5 seconds, or am I over-thinking this?